torchgeo/datasets/mdas.py

```diff
@@ -0,0 +1,347 @@
+    MDAS supports applications like resolution enhancement, spectral unmixing, and land cover classification.
+
+    Dataset features:
+
+    * 3K DSM data
+    * 3K high resolution RGB image
+    * Original very high resolution HySpex airborne imagery
+    * EeteS simulated imagery with 10m GSD and EnMAP spectral bands
+    * EeteS simulated imagery with 30m GSD and EnMAP spectral bands
+    * EeteS simulated imagery with 10m GSD and Sentinel-2 spectral bands
+    * Sentinel-2 L2A product
+    * Sentinel-1 GRD product
+    * Open Street Map (OSM) labels
+
+    Dataset format:
+
+    * 3K_RGB.tif (Shape: (4, 15000, 18000)px, Data Type: uint8)
+    * 3K_dsm.tif (Shape: (1, 10000, 12000)px, Data Type: float32)
+    * HySpex.tif (Shape: (368, 1364, 1636)px, Data Type: int16)
+    * EeteS_EnMAP_2dot2m.tif (Shape: (242, 1364, 1636)px, Data Type: float32)
+    * EeteS_EnMAP_10m.tif (Shape: (242, 300, 360)px, Data Type: uint16)
+    * EeteS_EnMAP_30m.tif (Shape: (242, 100, 120)px, Data Type: uint16)
+    * EeteS_Sentinel_2_10m.tif (Shape: (4, 300, 360)px, Data Type: uint16)
+    * Sentinel_2.tif (Shape: (12, 300, 360)px, Data Type: uint16)
+    * Sentinel_1.tif (Shape: (2, 300, 360)px, Data Type: float32)
+    * osm_buildings.tif (Shape: (1, 1364, 1636)px, Data Type: uint8)
+    * osm_landuse.tif (Shape: (1, 1364, 1636)px, Data Type: float64)
+    * osm_water.tif (Shape: (1, 1364, 1636)px, Data Type: float64)
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://essd.copernicus.org/articles/15/113/2023/
+
+    .. versionadded:: 0.7
+    """
+
+    valid_modalities = (
+        '3K_DSM',
+        '3K_RGB',
+        'HySpex',
+        'EeteS_EnMAP_10m',
+        'EeteS_EnMAP_30m',
+        'EeteS_Sentinel_2_10m',
+        'Sentinel_2',
+        'Sentinel_1',
+        'osm_buildings',
+        'osm_landuse',
+        'osm_water',
+    )
+    landuse_class_names: ClassVar[dict[int, str]] = {
+        0: 'no label',
+        1: 'forest',
+        2: 'park',
+        3: 'residential',
+        4: 'industrial',
+        5: 'farm',
+        6: 'cemetery',
+        7: 'allotments',
+        8: 'meadow',
+        9: 'commercial',
+        10: 'nature reserve',
+        11: 'recreation ground',
+        12: 'retail',
+        13: 'military',
+        14: 'quarry',
+        15: 'orchard',
+        16: 'scrub',
+        17: 'grass',
+        18: 'heath',
+    }
+
+    landuse_mapping: ClassVar[dict[int, int]] = {
+        -2147483647: 0,
+        7201: 1,
+        7202: 2,
+        7203: 3,
+        7204: 4,
+        7205: 5,
+        7206: 6,
+        7207: 7,
+        7208: 8,
+        7209: 9,
+        7210: 10,
+        7211: 11,
+        7212: 12,
+        7213: 13,
+        7214: 14,
+        7215: 15,
+        7217: 16,
+        7218: 17,
+        7219: 18,
+    }
+
+    cmap = ListedColormap([plt.cm.tab20(i) for i in range(20)])  # type: ignore[attr-defined]
```

Comment:
I don't love ignoring the original type hints here. We may want to think about whether we want to change `cmap` to always be some kind of matplotlib-compatible cmap or keep it as a rasterio-compatible cmap. For now, let's keep it in the plot method so we don't have to silence mypy.